docs/user-guide/creating-application/fluxcd.md

```diff
@@ -0,0 +1,118 @@
+# GitOps (Flux CD)
+ 
+{% hint style="info" %} 
+### Prerequisite
+
+Please make sure to install **Build and Deploy (CI/CD)** integration. To install it, click [here](../integrations/build-and-deploy-ci-cd.md).
+
+{% endhint %}
+
+Devtron supports FluxCD to help you manage application deployments using GitOps. With FluxCD, your Git repository becomes the single source of truth for your Kubernetes apps. Any changes you make in Git are automatically applied to your Kubernetes cluster by FluxCD. Refer to the [FluxCD documentation](https://fluxcd.io/flux/) to learn more.
+
+{% hint style="info" %}
+### Additional Resources
+ [What is FluxCD? A Quick Guide to GitOps with FluxCD](https://devtron.ai/blog/what-is-fluxcd/)
+
+ [Choosing the Right GitOps Tool: ArgoCD vs.FluxCD](https://devtron.ai/blog/gitops-tool-selection-argo-cd-or-flux-cd/)
+
+{% endhint %}
+
+## Key features
+
+* Works with Devtron CI pipelines and other Devtron integrations.
+
+* Supports Helm deployment method only.
+
+## Installation
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+The user must have permissions to:
+  * Edit the ConfigMaps of 'default-cluster'
+  * Restart the pods
+{% endhint %}
+
+To enable deployments through GitOps via FluxCD or if you want to migrate your existing FluxCD application to Devtron, you need to enable two specific feature flags for the `default_cluster` in **Devtron** and then install the FluxCD controller on those clusters in which you want to deploy the FluxCD applications. To do so, follow the steps below:
```

Comment:
The mentioned 2 steps shouldn't be inter dependent, I can choose to enable new flux deployments but not migrate and vice versa